src/handleAction.js

```diff
@@ -1,28 +1,22 @@
-function isFunction(val) {
-  return typeof val === 'function';
-}
+import isFunction from 'lodash.isfunction';
+import identity from 'lodash.identity';
+import isNil from 'lodash.isnil';
+import isSymbol from 'lodash.issymbol';
 
 export default function handleAction(type, reducers, defaultState) {
-  const typeValue = isFunction(type)
-    ? type.toString()
-    : type;
-
-  return (state = defaultState, action) => {
-    // If action type does not match, return previous state
-    if (action.type !== typeValue) return state;
+  const typeValue = isSymbol(type)
+    ? type
+    : type.toString();
 
-    const handlerKey = action.error === true ? 'throw' : 'next';
+  const [nextReducer, throwReducer] = isFunction(reducers)
+    ? [reducers, reducers]
+    : [reducers.next, reducers.throw].map(reducer => (isNil(reducer) ? identity : reducer));
 
-    // If function is passed instead of map, use as reducer
-    if (isFunction(reducers)) {
-      reducers.next = reducers.throw = reducers;
+  return (state = defaultState, action) => {
+    if (action.type !== typeValue) {
+      return state;
     }
 
-    // Otherwise, assume an action map was passed
-    const reducer = reducers[handlerKey];
-
-    return isFunction(reducer)
-      ? reducer(state, action)
-      : state;
+    return (action.error === true ? throwReducer : nextReducer)(state, action);
```

Comment:
You are right, `error` could be any other value than `true`. `=== true` adds more safety to respect FSA.